### PR TITLE
 Volume_planes in Polyhedron_demo : Fix for 775

### PR DESCRIPTION
**Pull request by maxGimeno**
This is a fix for #775 

This PR : 
- Puts the planes and intersection items generated by the plugin into a group_item (one per segmented_image_item). 
- Restores the possibility to pick a plane with shift+click, and updates the isovalue displayed by the plugin when done so.
- Makes that only 3 slides are displayed at the same time, and link them to the active segmented_image_item.
- Modifies the display of the intersection item, so that it is not hidden, nor partially hidden, by the planes (the intersection is displayed on top of the polygons).
- Adds a dialog displaying the numbers of planes currently created during planes creation. This dialog's true role is to keep the user from deleting the item or the group before all threads are finished, which would cause a segfault and make the demo crash. Note that there is no possibility to cancel a planes creation and so, once it is started, if the user wants to stop, he/she will not have access to the exit cross of the window, and will have to stop the process.

### CODE REVIEW

**Comment by janetournois** [comment]
I am wondering what happens when one tries to have volume planes on more than one image. There should be 2 groups of items, and the sliders should be connected to the one which is selected.

For only one image, this PR seems to solve the issue. Thanks!

**Comment by afabri** [comment]
I would suggest that dealing with one image is sufficient.

**Comment by maxGimeno** [comment]
If one makes planes for several images one after another, it should work. I
did nothing to produce planes for several images at the same time though.

2016-02-26 18:29 GMT+01:00 Andreas Fabri <notifications@github.com>:

**Comment by lrineau** [comment]
For the labelized images with weight we use two images: the labels images, and the weights image. If the demo can only use one image at a time, that will limit us.

**Comment by maxGimeno** [comment]
Yes but as long as the planes don't have to be created in the same time, there shouldn't be any problem. I tested with 2 images at the same time, it works well.

**Comment by janetournois** [comment]
Planes are created properly, but how should they be manipulated using the slide bars? The "active" item should be the one that is manipulated by the xyz sliders.

**Comment by janetournois** [comment]
Thanks @maxGimeno for help. The image item that is selected is the one that gets cut by the planes.

**Comment by maxGimeno** [comment]
I just added a little text to clarify that, along with a fix for the segfault we detected this morning.

**Comment by janetournois** [comment]
great, thanks

**Comment by lrineau** [comment]
@maxGimeno Please rebase (with a real `git rebase`) your branch on `master`.

Maybe you can also rework the history. See that help page from GitHub: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

**Comment by maxGimeno** [comment]
It's done

**Comment by lrineau** [comment]
Nice work. The history of the PR looks really good, now.

This PR will be tested in [CGAL-4.9-Ic-7][i]...

[i]: https://cgal.geometryfactory.com/CGAL/testsuite/results-4.9-Ic-7.shtml

File: Three/include/CGAL/Three/Viewer_interface.h [hunk 1]
```diff
@@ -136,6 +136,7 @@ class VIEWER_EXPORT Viewer_interface : public QGLViewer, public QOpenGLFunctions
   void requestContextMenu(QPoint global_pos);
   //!Defined automatically in moc.
   void selectedPoint(double, double, double);
+  void pointSelected(QPoint p);
   //!Defined automatically in moc.
```
Comment (lrineau) [inline Three/include/CGAL/Three/Viewer_interface.h]: Should not `pointSelected` renamed to `pixelSelected`?

Comment (lrineau) [inline Three/include/CGAL/Three/Viewer_interface.h]: By the way, the base class `QGLViewer` already has:
```
void pointSelected 	( 	const QMouseEvent *  	e	)
```
that does almost the same thing: the `QPoint p` is `e->pos()`.

I really would prefer that the `.h` of the framework are only modified when necessary, because that may mean:
  - recompilations
  - or ABI-incompatibilities.

Comment (maxGimeno) [inline Three/include/CGAL/Three/Viewer_interface.h]: It's a possibility, but a 2D Point is also a point, isn't it ?

Comment (lrineau) [inline Three/include/CGAL/Three/Viewer_interface.h]: Right, both are points, a 2D point and a 3D point. But with those names, I am curious to see how you will document the two functions. A name with "pixel" would lift the ambiguity.

File: Polyhedron/demo/Polyhedron/Plugins/PMP/Selection_widget.ui [hunk 2]
```diff
@@ -298,42 +298,44 @@
...
+      </item>
+      <item>
+       <widget class="QPushButton" name="validateButton">
+        <property name="text">
+         <string>Validate</string>
+        </property>
+       </widget>
+      </item>
+     </layout>
     </item>
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Selection_widget.ui]: Why is this file modified by this PR?!

Comment (maxGimeno) [inline Polyhedron/demo/Polyhedron/Plugins/PMP/Selection_widget.ui]: It shouldn't.

**Comment by lrineau** [comment]
There is a warning, on Linux Fedora:
> demo/Polyhedron/Plugins/Mesh_3/Volume_planes_plugin.cpp:324:14: warning: variable ‘control’ set but not used [-Wunused-but-set-variable]
